jerry-core/debugger/jerry-debugger.c

```diff
@@ -1 +1,113 @@
+int connected;                /**< hold the file descriptor for the accepted socket */
+
+struct sockaddr_in server_addr, client_addr;  /**< declarations of the socket address */
+socklen_t sin_size = sizeof (struct sockaddr_in);  /**< size of the structure pointed by
+                                                    *   the server_addr and client_addr */
+
+bool remote_init ()
+{
+  /* The arguments optval is used to access option values for setsockopt(). */
+  bool optval = true;
+
+  /* Server adress declaration */
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_port = htons (PORT);
+  server_addr.sin_addr.s_addr = INADDR_ANY;
+  bzero (&(server_addr.sin_zero), BACKLOG);
+
+  /* Create an endpoint for communication */
+  if ((sock = socket (AF_INET, SOCK_STREAM, 0)) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", strerror (errno));
+    return false;
+  }
+
+  /* Set the options on socket */
+  if (setsockopt (sock, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof (int)) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Erorr: %s\n", strerror (errno));
+    return false;
+  }
+
+  /* Bind to the server address */
+  if (bind (sock, (struct sockaddr *)&server_addr, sizeof (struct sockaddr)) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", strerror (errno));
+    return false;
+  }
+
+  /* Listen for connections on socket */
+  if (listen (sock, BACKLOG) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", strerror (errno));
+    return false;
+  }
+
+  jerry_port_log (JERRY_LOG_LEVEL_DEBUG, "Waiting for the client connection.\n");
+
+  /* Connect from the client */
+  if ((connected = accept (sock, (struct sockaddr *)&client_addr, &sin_size)) == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", strerror (errno));
+    return false;
+  }
+
+  jerry_port_log (JERRY_LOG_LEVEL_DEBUG, "Connected from: %s:%d\n",
+                  inet_ntoa (client_addr.sin_addr), ntohs (client_addr.sin_port));
+
+  return true;
+} /* remote_init */
+
+/* Close the socket connection with the client */
+void connection_closed ()
+{
+  jerry_port_log (JERRY_LOG_LEVEL_DEBUG, "TCPServer connection closed on port: %d\n", PORT);
+  close (connected);
+  close (sock);
+} /* connection_closed */
+
+/* Send the parsed file names to the client side */
+void send_to_client (uint16_t data_len) /**< data length */
+{
+  ssize_t byte_send = send (connected, jerry_debugger_buffer, data_len, 0);
+
+  /* If send returns -1 then some error messages are generated by the socket layer. */
+  if (byte_send == -1)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", strerror (errno));
+  }
+  /* Compare the byte_send size with the data_len. If it is not equal then not all of the
+   * information is sent to the client. */
+  else if (byte_send != (ssize_t) data_len)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: Missing bytes from the pacakge.\n");
```

Comment:
package.
Ot just say: Error: transmission error.

Btw, I think a loop would be better here to send the remaining bytes. Only -1 is a true error, partial transmission could be resumed.